source/content/guides/secrets/06-troubleshooting.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Pantheon Secrets Guide
+subtitle: Troubleshooting
+description: Securely store secrets in the Pantheon Platform.
+terminuspage: true
+contributors: [stovak]
+contenttype: [guide]
+innav: [true]
+categories: [secrets]
+cms: [drupal, wordpress]
+audience: [development]
+product: [secrets]
+integration: [--]
+tags: [reference, cli, local, terminus, workflow]
+permalink: docs/guides/secrets/troubleshooting
+reviewed: "2024-05-01"
+---
+## Integrated Composer Build fails on private packages
+
+This is the most common error when sites are using secrets and Integrated Composer. This may manifest in different ways and may be caused by different problems. This playbook tries to cover them.
+
+### Error getting a private package during the IC build
+
+This is the most common error; an example message associated to this error is:
+
+```
+Failed to download vendor/package from dist
+```
+(where vendor/package may be any private package)
+
+Some possible causes for this error:
+
+- **Problem:** Secrets are not correctly set for the site. Secrets for Integrated Composer to use need to be type `composer` and have scope `ic`. Secret types and scopes are covered in the [Basic Concepts](/guides/secrets/02-basic-concepts) documentation.
+
+  **Solution:** ask the client to delete and recreate the secret if scope and type do not match for the given secret name
+
+- **Problem:** Secret value/token may be expired
+
+  **Solution:** ask the client to set the secret again to an updated value
+
+- **Problem**: Site may be running on a PHP version below 7.4. If this is the case, there will be a message in the job output: “Skipping setting up secrets as it is not supported in PHP below 7.4”
```

Comment:
```suggestion
- **Problem**: Site may be running on a PHP version below 8.0. If this is the case, there will be a message in the job output: “Skipping setting up secrets as it is not supported in PHP below 8.0”
```
[PHP 8.0 or later is required](https://github.com/pantheon-systems/site-job-scripts/blob/aa42f269e00747a9d7a4d1a7b9daafa548edfbfc/devops/k8s/configmaps/all/job-scripts/setup-secrets.php#L15) to use Secrets in IC builds.

Reply:
Someone requested this 8.0 be changed to 7.4, but I can't remember who, and I'm not sure how to search through the nearly 100 conversations to find which one was about PHP version support.

Secrets in IC builds are not supported in PHP 7.4: https://github.com/pantheon-systems/site-job-scripts/blob/master/devops/k8s/configmaps/all/job-scripts/setup-secrets.php#L12-L16

I think only the terminus plugin and the appserver runtime library work in PHP 7.4.  IC does not, and this section is about IC.

cc @jazzsequence in case it was you that requested this be changed to 7.4.